src/core_plugins/kibana/public/management/sections/indices/add_data_steps/parse_csv_step/lib/validate_headers.js

```diff
@@ -0,0 +1,41 @@
+import _ from 'lodash';
+
+export default function validateHeaders(fields) {
+  if (!_.isArray(fields)) throw new Error('first argument must be an array');
+
+  let emptyPositions = [];
+  let errors = [];
+
+  _.forEach(fields, (field, index, interatee) => {
+    // Check if field is empty
+    if (_.isEmpty(field)) {
+      emptyPositions.push(index + 1);
+    }
+    // Skip if this header is already checked
+    else if (_.findIndex(errors, error => { return error.fieldName === field; }) < 0) {
+      let duplicatePositions = [];
+      // Find other headers with the same name
+      for (let i = index; i < fields.length; i++) {
+        if (field === fields[i]) {
+          duplicatePositions.push(i + 1);
+        }
+      }
+      if (duplicatePositions.length > 1) {
+        errors.push({
+          'fieldName': field,
+          'positions': duplicatePositions,
+          'type': 'duplicate'
+        });
+      }
+    }
```

Comment:
I think the logic could be simplified here by using lodash's `reduce`. Your accumulator would be an object keyed by field name with error objects as values, and after the reduce you could simply apply a filter that removes any error objects with postitions === 1. JS even allows empty strings as object keys, so you can probably handle the empty header case at the same time.